src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.CreateDefaultHandler.Android.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+
+namespace System.Net.Http
+{
+    public partial class HttpClient
+    {
+        private const string HandlerSwitchName = "System.Net.Http.UseNativeHttpHandler";
+
+        private static MethodInfo? handlerMethod;
+
+        private static HttpMessageHandler CreateDefaultHandler()
+        {
+            // Default is to use the Android native handler
+            if (!AppContext.TryGetSwitch(HandlerSwitchName, out bool isEnabled))
```

Comment:
This has to be private static property/method which you could write substitution. See how it's done for `System.Net.Http.DiagnosticsHandler. IsGloballyEnabled` https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Http/src/ILLink/ILLink.Substitutions.xml

I'd also consider placing that code in a shared file, it does not matter that it's not used on desktop